src/python/pants/init/options_initializer.py

```diff
@@ -105,18 +105,18 @@ def _install_options(self, options_bootstrapper, build_configuration):
     from pants.bin.goal_runner import GoalRunner
 
     # Now that plugins and backends are loaded, we can gather the known scopes.
-    known_scope_infos = [GlobalOptionsRegistrar.get_scope_info()]
+    known_scope_infos = set(GlobalOptionsRegistrar.known_scope_infos())
 
     # Add scopes for all needed subsystems via a union of all known subsystem sets.
     subsystems = Subsystem.closure(
       GoalRunner.subsystems() | Goal.subsystems() | build_configuration.subsystems()
     )
-    for subsystem in subsystems:
-      known_scope_infos.append(subsystem.get_scope_info())
+    for ss in subsystems:
+      known_scope_infos |= set(ss.known_scope_infos())
```

Comment:
Would generally recommend `known_scope_infos.update(ss.known_scope_infos())`... haven't benchmarked it, but it is one fewer operation (just the add to set operation, without an intersect afterward).

Reply:
That makes sense, I've made the change.